wallet/src/actors/worker/methods.rs

```diff
@@ -1063,7 +1067,13 @@ impl Worker {
     ) -> Result<bool> {
         // Only trigger sync of chain if data has been cleared
         match wallet.clear_chain_data()? {
-            true => self.sync(wallet_id, wallet, sink, true).map(|_| true),
+            true => {
+                wallet.apply_on_state(|mut state| state.synchronization.set_resyncing(true))?;
```

Comment:
Same as above, shouldn't this check if that flag is already set to true, and in that case return some error?

Reply:
That's checked inside `wallet.clear_chain_data`. What do you think about moving that guard outside so that `clear_chain_data` doesn't need to care about the synchronization status?

Reply:
I don't know, it sounds like `clear_chain_data` should return an error if the wallet is syncing, or cancel the synchronization

Reply:
We can't do that. At least for the time being. Because that's exactly the issue we're having right now. `clear_chain_data` is blocking any resynchronization for wallets that are forked.